meteor/server/api/showStyles.ts

```diff
@@ -137,21 +157,47 @@ export async function removeShowStyleVariant(
 	await ShowStyleVariants.removeAsync(showStyleVariant._id)
 }
 
+export async function reorderAllShowStyleVariants(
+	context: MethodContext,
+	showStyleBaseId: ShowStyleBaseId,
+	orderedVariants: ShowStyleVariant[]
+): Promise<void> {
+	await assertShowStyleBaseAccess(context, showStyleBaseId)
+
+	await reorderAllShowStyleVariantsInner(orderedVariants)
+}
+
+export async function reorderAllShowStyleVariantsInner(orderedVariants: ShowStyleVariant[]): Promise<void> {
+	orderedVariants.forEach((variant: ShowStyleVariant, index: number) => {
+		ShowStyleVariants.upsert(variant._id, {
+			$set: {
+				_rank: index,
+			},
+		})
+	})
+}
+
 class ServerShowStylesAPI extends MethodContextAPI implements NewShowStylesAPI {
 	async insertShowStyleBase() {
 		return insertShowStyleBase(this)
 	}
-	async insertShowStyleVariant(showStyleBaseId: ShowStyleBaseId) {
-		return insertShowStyleVariant(this, showStyleBaseId)
+	async createDefaultShowStyleVariant(showStyleBaseId: ShowStyleBaseId, rank: number) {
```

Comment:
I changed this in the coming commit, since it also took care of the other issue you raised about ensuring that the highest rank is alway provided to new default variants.